app/models/project.js

```diff
@@ -85,21 +75,25 @@ var Project = Resource.extend(PolledResource, {
     let l = this.get('links');
 
     var choices = [
-      { label: 'action.setDefault',       icon: 'icon icon-star-fill',    action: 'setAsDefault', enabled: this.get('canSetDefault')},
-      { divider: true },
-      { label: 'action.edit',             icon: 'icon icon-edit',         action: 'edit',         enabled: !!l.update },
-      { divider: true },
-      { label: 'action.activate',         icon: 'icon icon-play',         action: 'activate',     enabled: !!a.activate, bulkable: true},
-      { label: 'action.deactivate',       icon: 'icon icon-pause',        action: 'promptStop',   enabled: !!a.deactivate, altAction: 'deactivate', bulkable: true},
+      { label: 'action.edit',             icon: 'icon icon-edit',         action: 'edit',         enabled: true},
       { divider: true },
-      { label: 'action.remove',           icon: 'icon icon-trash',        action: 'promptDelete', enabled: !!l.remove, altAction: 'delete', bulkable: true },
+      { label: 'action.remove',           icon: 'icon icon-trash',        action: 'promptDelete', enabled: true, altAction: 'delete', bulkable: true },
       { divider: true },
       { label: 'action.viewInApi',        icon: 'icon icon-external-link',action: 'goToApi',      enabled: true },
     ];
 
     return choices;
   }),
 
+  delete: function (/*arguments*/) {
+    var promise = this._super.apply(this, arguments);
+    return promise.then(() => {
+      this.set('state', 'removed');
+    }).catch((err) => {
+      this.get('growl').fromError('Error deleting', err);
+    });
+  },
```

Comment:
Something still probably needs to happen to handle the case where you delete the project that was last selected, since the subscribe connection will still be active..  Reloading was a simple way to handle it for this edge-ish case.